src/Models/Hooks/useTimeSeriesData.ts

```diff
@@ -0,0 +1,202 @@
+            adapter,
+            connectionString,
+            twins,
+            pollingInterval
+        ],
+        isAdapterCalledOnMount: false,
+        isLongPolling: pollingInterval ? true : false,
+        pollingIntervalMillis: pollingInterval
+    });
+
+    const onMountDate = useMemo(() => Date.now(), []);
+
+    /**
+     * After getting time series data from adapter, parse it in a structure of
+     * [timestamp, value] and store it as data to expose for consumers of this hook
+     *  */
+    useEffect(() => {
+        if (timeSeriesData?.adapterResult?.result?.data) {
+            logDebugConsole(
+                'debug',
+                `[END]: Fetching time series data finished: `,
+                timeSeriesData?.adapterResult?.result?.data
+            );
+            setData(timeSeriesData?.adapterResult?.result?.data);
+        } else {
+            setData(null);
+        }
+    }, [timeSeriesData?.adapterResult.result]);
+
+    useEffect(() => {
+        if (connectionString) {
+            setConnectionToQuery(
+                generateADXConnectionFromString(connectionString)
+            );
+        } else {
+            setConnectionToQuery(adapter?.getADXConnectionInformation());
+        }
+    }, [connectionString, adapter]);
+
+    const prevQuery = usePrevious(query);
+    useEffect(() => {
+        let newQuery = '';
+        if (connectionToQuery && quickTimeSpan && twins?.length) {
+            const timeFrom = new Date(onMountDate - quickTimeSpan);
+            const timeTo = new Date(onMountDate);
+            newQuery = getBulkADXQueryFromTimeSeriesTwins(
+                twins,
+                { from: timeFrom, to: timeTo },
+                connectionToQuery
+            );
+        }
+        if (prevQuery !== newQuery) {
+            setQuery(newQuery);
+        }
+    }, [connectionToQuery, quickTimeSpan, twins, onMountDate]);
+
+    const prevDeeplink = usePrevious(deeplink);
+    useEffect(() => {
+        const newDeeplink = getADXDeeplinkForWeb(connectionToQuery, query);
+        if (prevDeeplink !== newDeeplink) {
+            setDeeplink(newDeeplink);
+        }
+    }, [connectionToQuery, query]);
+
+    const fetchData = () => {
+        logDebugConsole(
+            'debug',
+            `[START]: Fetching time series data using query "${query}"`
+        );
+        timeSeriesData.callAdapter({
+            query,
+            connection: connectionToQuery
+        });
+    };
+
+    return useMemo(() => {
+        return {
+            query,
+            deeplink,
+            data,
+            fetchTimeSeriesData: fetchData,
+            isLoading: timeSeriesData.isLoading
+        };
+    }, [query, deeplink, data, fetchData, timeSeriesData.isLoading]);
+};
+
+const generateADXConnectionFromString = (
+    connectionString: string
+): IADXConnection | null => {
+    if (connectionString) {
+        try {
+            const parts = connectionString.split(';');
+            return {
```

Comment:
how come we store the connection string as a single string if we need it split up like this?

Reply:
the syntax of connection string here is a common practice like key=value;key2=value2.... That is why it is a single string value in config file, but we need to deconstruct it to get proper fields from it.

Reply:
I undersatnd that's standard practice for most connection strings, but usually that's how they are used. In our case, we don't use it that way. So why do we go through the trouble to make it into a string only to deconstruct it again later?

Reply:
what is your suggestion? in the config schema how can we keep those keys (cluster url, database name, table name)? since in the future it might be arbitrary keys depending on the data explorer service if we keep those separately in the schema, we will need to update the schema.